### PR TITLE
adding osxarm

### PR DESCRIPTION
**Pull request by ngam**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@asford @sodre

I have added osx-arm64 support here. For some reason, `go install` didn't work and instead I had to use `go build`. I am not really sure if this fundamentally changes anything --- there are some warnings in the log --- but it looks like it works fine. Please have a look and merge, thanks!

```diff
- go install -v .
+ # go install -v .
+
+ if [[ "$target_platform" = "osx-arm64" ]]; then
+   go build -v .
+ fi
+
+ if [[ "$target_platform" != "osx-arm64" ]]; then
+   go install -v .
+ fi
```

**Comment by ngam** [comment]
> @asford @sodre
> 
> I have added osx-arm64 support here. For some reason, `go install` didn't work and instead I had to use `go build`. I am not really sure if this fundamentally changes anything --- there are some warnings in the log --- but it looks like it works fine. Please have a look and merge, thanks!
> 
> ```diff
> - go install -v .
> + # go install -v .
> +
> + if [[ "$target_platform" = "osx-arm64" ]]; then
> +   go build -v .
> + fi
> +
> + if [[ "$target_platform" != "osx-arm64" ]]; then
> +   go install -v .
> + fi
> ```

@matthiasdiener does this sound reasonable to make osxarm pass?

**Comment by matthiasdiener** [comment]
> > @asford @sodre
> > I have added osx-arm64 support here. For some reason, `go install` didn't work and instead I had to use `go build`. I am not really sure if this fundamentally changes anything --- there are some warnings in the log --- but it looks like it works fine. Please have a look and merge, thanks!
> > ```diff
> > - go install -v .
> > + # go install -v .
> > +
> > + if [[ "$target_platform" = "osx-arm64" ]]; then
> > +   go build -v .
> > + fi
> > +
> > + if [[ "$target_platform" != "osx-arm64" ]]; then
> > +   go install -v .
> > + fi
> > ```
> 
> @matthiasdiener does this sound reasonable to make osxarm pass?

I don't know much about cross-compiling go, sorry :-(

**Comment by ngam** [comment]
Thanks. @conda-forge/core, could you please merge this?

**Comment by matthiasdiener** [comment]
Just to make sure @ngam, did you check that the osxarm package contains the actual files necessary to run `hub`?

**Comment by matthiasdiener** [comment]
AFAICS, the difference between the osx64 and osxarm CI outputs seem to indicate the package is built, but *not* installed correctly. Please double-check.

**Comment by ngam** [comment]
Okay, let me double check and test locally before merging. 

Upstream, they seem to use `go build`, so I think this works, but I will do a little more testing locally before we move forward.

https://github.com/github/hub/blob/master/script/build

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@matthiasdiener thanks a lot. I tried copying the executables but was faced with a problem with the flags. I was partially inspired by your work in https://github.com/conda-forge/gh-feedstock/blob/master/recipe/build.sh

This works fine locally, but please have a look and let me know if it makes sense to you and if you have any idea about the flags...

**Comment by ngam** [comment]
> I don't know much about cross-compiling go, sorry :-(

@matthiasdiener you know a lot! Thanks for spotting the mistake here. I added an install statement and that fixes everything locally and passes here. Now the build process is identical for osx and osx-arm, except the tests which are skipped in the case of osx-arm.

Got inspired by your excellent work here: https://github.com/conda-forge/gh-feedstock/blob/master/recipe/build.sh

Thanks a lot!!

**Comment by ngam** [comment]
@conda-forge/hub @conda-forge/core please merge, thanks!

File: recipe/build.sh [hunk 1]
```diff
@@ -4,4 +4,14 @@ set -eux
 mkdir -p $PREFIX/etc/bash_completion.d;
 cp etc/hub.bash_completion.sh $PREFIX/etc/bash_completion.d/hub.sh
 
-go install -v .
+# go install -v .
+
+if [[ "$target_platform" = "osx-arm64" ]]; then
```
Comment (wolfv) [inline recipe/build.sh]: shouldn't this be `==`?

File: recipe/build.sh [hunk 2]
```diff
@@ -4,4 +4,14 @@ set -eux
 mkdir -p $PREFIX/etc/bash_completion.d;
 cp etc/hub.bash_completion.sh $PREFIX/etc/bash_completion.d/hub.sh
 
-go install -v .
+# go install -v .
```
Comment (wolfv) [inline recipe/build.sh]: maybe remove this line?

File: recipe/build.sh [hunk 3]
```diff
@@ -4,4 +4,14 @@ set -eux
...
+
+if [[ "$target_platform" = "osx-arm64" ]]; then  
+  go build -v .
+  mkdir -p ${PREFIX}/bin
+  install hub ${PREFIX}/bin
```
Comment (wolfv) [inline recipe/build.sh:10]: are these all files that need to be installed?

**Comment by wolfv** [comment]
looks good, maybe squash the commit history a bit?

File: recipe/build.sh (hunk 1 above)
Comment (ngam) [reply recipe/build.sh]: Yes, will fix now.

File: recipe/build.sh (hunk 3 above)
Comment (ngam) [reply recipe/build.sh:10]: I believe so, but let me double check locally first.

File: recipe/build.sh (hunk 2 above)
Comment (ngam) [reply recipe/build.sh]: Done.

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh (hunk 3 above)
Comment (ngam) [reply recipe/build.sh:10]: Yes, all good.

**Comment by ngam** [comment]
@wolfv should be good to go, thanks a lot!

**Comment by ngam** [comment]
@conda-forge-admin, restart ci

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
travis ci isn't cooperating... so changed it to azure.

@conda-forge-admin, please rerender

**Comment by matthiasdiener** [comment]
> > I don't know much about cross-compiling go, sorry :-(
> 
> @matthiasdiener you know a lot! Thanks for spotting the mistake here. I added an install statement and that fixes everything locally and passes here. Now the build process is identical for osx and osx-arm, except the tests which are skipped in the case of osx-arm.
> 
> Got inspired by your excellent work here: https://github.com/conda-forge/gh-feedstock/blob/master/recipe/build.sh
> 
> Thanks a lot!!

Great, glad this was helpful! :-)

**Comment by ngam** [comment]
@matthiasdiener would you like to be a maintainer as well here since you are maintainer of gh? Happy to add you now that I am a maintainer, just let me know. Thanks a lot!

**Comment by matthiasdiener** [comment]
> @matthiasdiener would you like to be a maintainer as well here since you are maintainer of gh? Happy to add you now that I am a maintainer, just let me know. Thanks a lot!

Sure, please feel free to add me!

**Pull request merged**